### PR TITLE
[Azure Search] Add URL encode + decode field mapping functions

### PR DESCRIPTION
**Pull request by arv100kri**
Add the `urlEncode` and `urlDecode` field mapping functions, that were recently added to the back end service. These are not swagger reliant and are wholly introduced via customizations

### CODE REVIEW

**Review by Careyjmac — APPROVED** [review]
I honestly don't know too much about this specific change, but overall it looks good to me.

File: sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Indexers/Models/FieldMappingFunction.Customization.cs [hunk 1]
```diff
@@ -44,6 +44,31 @@ public partial class FieldMappingFunction
                     [nameof(useHttpServerUtilityUrlTokenEncode)] = useHttpServerUtilityUrlTokenEncode
                 });
 
+        /// <summary>
+        /// Creates a field mapping function that performs a simple URL-safe encoding of the input string.
+        /// It uses UTF-8 encoding and utilizes HttpUtility's UrlEncode method.
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Indexers/Models/FieldMappingFunction.Customization.cs]: Should we be leaking server-side implementation details here?

File: sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Indexers/Models/FieldMappingFunction.Customization.cs [hunk 2]
```diff
@@ -44,6 +44,31 @@ public partial class FieldMappingFunction
...
+        /// See <see cref="https://docs.microsoft.com/dotnet/api/system.web.httputility.urlencode"/> for more details.
+        /// </summary>
+        /// <remarks>
+        /// Sample use case: This field mapping function can be used as an alternative to Base64Encode if only the URL
+        /// unsafe characters of a key field need to be safely converted, while other characters are to remain as is.
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Indexers/Models/FieldMappingFunction.Customization.cs]: nit: as-is

**Review by bitsforthought — CHANGES_REQUESTED** [review]
We should find a platform-independent way to describe the functionality of these field mapping functions. While in this case it's .NET developers reading the docs, what about for our future Java SDK? I would argue we shouldn't be leaking implementation details of our REST API in SDK docs.

File: sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Indexers/Models/FieldMappingFunction.Customization.cs (hunk 1 above)
Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Indexers/Models/FieldMappingFunction.Customization.cs]: Fair enough - I'll first update our documentation for field mapping functions (similar to base64encode and decode) and then refer to that link here.

I'll hold off on this PR, until the documentation is out at which point I'll send a new commit updating the docs

Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Indexers/Models/FieldMappingFunction.Customization.cs]: On second thought, I think URL-encoding + specifying the encoding type should be generic enough and useful as-is and doesn't require serializing behind a documentation update - new commit incoming :)

**Comment by arv100kri** [comment]
> We should find a platform-independent way to describe the functionality of these field mapping functions. While in this case it's .NET developers reading the docs, what about for our future Java SDK? I would argue we shouldn't be leaking implementation details of our REST API in SDK docs.

Fixed in the latest commit!

**Review by bitsforthought — APPROVED** [review]

**Pull request merged**